### PR TITLE
Use ruff for CI linting

### PR DESCRIPTION
**Pull request by ljgray**
Similar to radiocosmology/caput#238

Also moves the unit tests for `invert_no_zero` to caput, since that method was moved previously. An associated PR can be found at radiocosmology/caput#240

### CODE REVIEW

File: draco/analysis/flagging.py [hunk 1]
```diff
@@ -132,10 +131,12 @@ def process(self, mmodes):
         Parameters
         ----------
         mmodes : containers.MModes
+            Mmode container to mask
 
         Returns
         -------
         mmodes : containers.MModes
+            Same object as input with maskig applied
```
Comment (jrs65) [inline draco/analysis/flagging.py:139]:
```suggestion
            Same object as input with masking applied
```

**Review by jrs65 — APPROVED** [review]
Looks good. Just spotted one typo.

File: draco/analysis/flagging.py (hunk 1 above)
Comment (ljgray) [reply draco/analysis/flagging.py:139]: Cool, I'll add that

**Comment by ljgray** [comment]
This PR seems to be breaking the sphinx autosummary...

**Pull request merged**